Sources/Testing/Issues/Issue.swift

```diff
@@ -115,9 +115,27 @@ public struct Issue: Sendable {
   @_spi(ForToolsIntegrationOnly)
   public var sourceContext: SourceContext
 
+  /// A type representing a
+  /// ``withKnownIssue(_:isIntermittent:sourceLocation:_:when:matching:)`` call
+  /// that matched an issue.
+  @_spi(ForToolsIntegrationOnly)
+  public struct KnownIssueContext: Sendable {
+    /// The comment that was passed to `withKnownIssue()`, if any.
+    var comment: Comment?
+
+    /// The source location that was passed to `withKnownIssue()`.
+    var sourceLocation: SourceLocation
```

Comment:
It's not clear to me why or how this source location is valuable.

Reply:
I was thinking that Xcode etc. could make it easy to navigate to the `withKnownIssue` call that matched this Issue. This seems helpful in the case of custom matcher functions, especially if you have multiple nested `withKnownIssue` calls. For example, when analyzing test results you might decide that a particular Issue should *not* be considered "known", and want to figure out which `withKnownIssue` call you need to adjust to address that.

Reply:
It is worth considering whether we should instead add a `sourceLocation` property to the `Comment` type instead. That's beyond the scope of this particular PR, I think, but it would then let us provide source location info for _all_ comments, not just this one, and in a way that's (ideally) more consistent.

Reply:
It's not the source location of the _comment_ that is interesting to capture, it's the source location of the `withKnownIssue()` call. And there may not always be a `Comment` passed to `withKnownIssue()`; it may be `nil`, so this `sourceLocation` property allows us to always capture the location of the `withKnownIssue()` call which caused an issue to be considered "known".

Reply:
Yeah, I can see the utility in that. Although I wonder if you'd ever want both: the place that constructed the `Comment` vs. the place that applied it to an `Issue`? I'm not certain that would be useful, but the two concepts do seem distinct to me, and a `sourceLocation` on `Comment` itself kind of implies the location of construction to me.

Reply:
Agree they're distinct. Personally I just find the location of a comment _less_ useful in comparison, and agreed, it's out of scope for this PR